internal/dependency/kv_exists_test.go

```diff
@@ -0,0 +1,353 @@
+package dependency
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/hcat/dep"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewKVExistsQuery_NonBlocking(t *testing.T) {
+	q, err := NewKVExistsQuery("key")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, ok := interface{}(q).(BlockingQuery); ok {
+		t.Fatal("should NOT be blocking")
+	}
+}
+
+func TestKVExistsQuery_SetOptions(t *testing.T) {
+	q, err := NewKVExistsQuery("key")
+	if err != nil {
+		t.Fatal(err)
+	}
+	q.SetOptions(QueryOptions{WaitIndex: 100, WaitTime: 100})
+	if q.opts.WaitIndex != 0 {
+		t.Fatal("WaitIndex should be 100")
```

Comment:
I might be a little confused here. The error message states that the WaitIndex should be 100 but the if-check is to check if it's not equal to 0 (similarly below). It seems as though the error message should be "WaitIndex should be 0". Let me know if I misunderstand!

Reply:
Oh, no, that's my bad, I meant to just update the error message for the KV Get test but it looks like I had changed both!

Reply:
Updated the error message and also added a comment to clarify that we do want to expect 0 even though the set options passed a non-zero value

Reply:
Nice! I was wondering about that